src/pages/annual_report/Report.js

```diff
@@ -162,179 +143,177 @@ const Report = function () {
 
   let selectedGroupInfo = (e) => {
     e.persist();
+    e.stopPropagation();
     setSelectedGroup(e.target.textContent);
     setOrgUnitGroupVisibility("none");
   };
 
   let selectedLevelInfo = (e) => {
     e.persist();
+    e.stopPropagation();
     setSelectedLevel(e.target.textContent);
-    setOrgUnitLevelVisibility("some information");
+    setOrgUnitLevelVisibility("none");
+  };
+  let [x, setX] = useState("x");
+  let [periodVisibility, setPeriodVisibility] = useState("none");
+  let [_dataGroupVisibility, _setDataGroupVisibility] = useState(false);
+  let [_orgUnitVisibility, _setOrgUnitVisibility] = useState(false);
+  let [_periodVisibility, _setPeriodVisibility] = useState(false);
+
+  const toggleSelectedLevel = (e) => {
+    e.persist();
+    setOrgUnitLevelVisibility((prev) => (prev === "none" ? "flex" : "none"));
+    setOrgUnitGroupVisibility("none");
+  };
+
+  const toggleSelectedGroup = (e) => {
+    e.persist();
+    setOrgUnitGroupVisibility((prev) => (prev === "none" ? "flex" : "none"));
+    setOrgUnitLevelVisibility("none");
   };
-  let [x, setX] = useState("x")
-  let [periodVisibility, setPeriodVisibility] = useState("none")
 
+  const deselectOrgUnitSelection = (e) => {
+    console.log("Deselection");
+  };
   return (
     <div className="reportContainer">
       <MenuBar />
-      <OrganizationUnitModal
-        status={orgUnitModalStatus}
-        changeOrganisationUnitStatus={setOrgUnitModalStatus}
-      />
-      <div className="menu-parent">
-        <div className="menu-parent-container">
-          <div className="data-set-container">
-            <div
-              className="dataset-indication"
-              onClick={() => setGroupVisibility("flex")}
+      <div className="for-selector-bar">
+        <SelectorBar
+          additionalContent={
+            <div className="additional-content">
+              <span>
+                <Button small primary>
+                  Generate report
+                </Button>
+              </span>
+              <span>
+                <Button small>Print</Button>
+              </span>
+            </div>
+          }
+          style={{ width: "800px" }}
+        >
+          <div>
+            <SelectorBarItem
+              label="Group"
+              value={selectedItem}
+              open={_dataGroupVisibility}
+              setOpen={() => _setDataGroupVisibility((prev) => !prev)}
```

Comment:
You're having trouble with the propagation because the setOpen function needs to take a boolean. For example, you should just pass `_setDataGroupVisibility` as the logic within the component sets the item to open/closed. Once you fix this, you should be able to use the components from the DHIS2 ui library and will not need to stop propagation in event handlers. 

I will open a ticket to improve the documentation for this component